### DISCUSSION TITLE
Is there a way to call GetLowestOfferListingsForASIN with SP API

### DISCUSSION THREAD

**Discussion by MyGuyCai** [saleweaver/python-amazon-sp-api#510] [FAQ]
As I've found in the MWS docs there is a call to get the lowest price for a list of asins. I want to achieve the same with selling partner api.

I have tried Products.get_competitive_pricing_for_asins which doesn't work with some products. The price goes higher than amazon's "competitive price" and doesn't return any prices.
I have also tried Products.get_product_pricing_for_asins which also doesn't work with some products. The returned payload is empty, even though I know the asin exists on that marketplace and there are offers.

Using this asin "B00GLBQ1PU" on the UK marketplace "A1F83G8C2ARO7P" with condition=used gives no competitive prices and no product prices.
If you search for the asin on amazon.co.uk the listing clearly exists and there are many used sellers.

Unless there is something obvious I'm missing is my only course of action is to call Products.get_item_offers on each ASIN one at a time? 
This is such a pain as it's 20x slower, turning a 1min30sec query into a 36min query

**Comment by IP050** [comment]
Hi, I don't have alot of experience using products(my usecases mostly lay around orders), but as I need to work with it soon figured il check out your problem.
I've checked using the asin u provided and did the same thing. get_pricing_for_asins does not return an empty payload for me:

payload': [{'ASIN': 'B00GLBQ1PU',
              'Product': {'Identifiers': {'MarketplaceASIN': {'ASIN': 'B00GLBQ1PU',
                                                              'MarketplaceId': 'A1F83G8C2ARO7P'}}},

But there is no information on price. I'm not sure if that is what u meant by it being 'empty'. What I am aiming at here is that when you have an empty payload it's probably the marketplaceid not provided (well enough).

For competitive_pricing_asins I got the same as you did(again, using your product). It gives a payload but the Competitive Prices are an empty list. So again no information on the price. 

You state that you did have some response in price for a product. If possible could you provide the ASIN that did return a price in competitive_prices_asins? Maybe there are differences product wise.

Other than that, for now it does look like that with these functions your only option in case above won't work is indeed to use get_item_offers. How many asins are you using that it will take up to 36min? Maybe we can speed up that process instead. 

Hope this somewhat helps, im happy to help you progress as it might have some value to me in the future as well, who knows.

edit: found this https://github.com/amzn/selling-partner-api-docs/issues/423 might be good to look into.

**Comment by MyGuyCai** [comment]
Hi, thanks for getting back to me.  
Yes, when I said 'empty' I meant that there was no price information in the payload. I am getting back the payload fine.  
I think the reason for an empty list being returned for some items with competitive_price is because amazon has a pre-definied competitive price range that does not update with the market value of products (I have had a few high-price warnings on my listings when my listing price was the lowest offer).  
You can try with this asin 'B000092BL6' that works with get_pricing_for_asins and also this asin 'B000LZ4M9Y' works with competitive_pricing_asins.
I have around 90,000 listings on the UK marketplace that I want to check to see if the value on the US marketplace is high enough for me to consider listing. As I have to take into account shipping costs, extra handling, higher non-deliveries etc
Thanks for posting that link, it seems they have the same issue I am facing.

**Reply by IP050** [reply]
Haha, with 90k asins the 36mins make more sense. No worries, let me know if the link helped you to solving your problem. 

I've tested the asins you provided, combining that with the post about the bug in getPricing I can see where the problem lays but can't find a solution by looking at the docs/models the seller-api contains. Any possible road seems unreliable/ slow to me. 

Just to confirm, Products.get_item_offers does give you all the information you need?

Another method I was thinking of might be scraping(although again, speed is probably an issue) or maybe a combination of both. Anyway, if you could update whenever you have the time on the current situation or if you found a solution I would appreciate it, your use-case is quite interesting to me.
